src/main/java/org/openstreetmap/atlas/checks/utility/ElevationUtilities.java

```diff
@@ -0,0 +1,361 @@
+import java.io.InputStream;
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Longitude;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * Used to determine assumed direction for some features (e.g. waterways). Assumptions are that any
+ * SRTM file follows the specification here:
+ * https://dds.cr.usgs.gov/srtm/version2_1/Documentation/Quickstart.pdf
+ *
+ * @author Taylor Smock
+ */
+public final class ElevationUtilities implements Serializable
+{
+    private static final long serialVersionUID = -5929570973909280629L;
+    /**
+     * The assumed file extension
+     */
+    private static final String SRTM_EXT = "hgt";
+    /**
+     * The assumed extent of a HGT SRTM file (lat/lon) in degrees
+     */
+    private static final int SRTM_EXTENT = 1;
+    /**
+     * A non-number when there is no elevation data available. This is currently returns
+     * {@link Short#MIN_VALUE}.
+     */
+    public static final short NO_ELEVATION = Short.MIN_VALUE;
+
+    private static final short[][] EMPTY_MAP = new short[][] {};
+
+    /** Just an int for converting a decimal to a percentage */
+    private static final int DECIMAL_TO_PERCENTAGE = 100;
+
+    /** A map of {lat, lon} pairs with a loaded srtm in a byte array */
+    private final Map<Pair<Integer, Integer>, short[][]> loadedSrtm = new HashMap<>();
+
+    private final int srtmExtent;
+
+    private final String srtmExt;
+
+    private final String srtmPath;
+
+    /**
+     * Configuration Keys in the Integrity Framework are based on the check simple classname.
+     *
+     * @param key
+     *            key part for a specific configuration item defined for this class
+     * @return complete key for lookup
+     */
+    private static String configurationKey(final String key)
+    {
+        return formatKey("elevationutilities", key);
+    }
+
+    private static <U, V> V configurationValue(final Configuration configuration, final String key,
+            final U defaultValue, final Function<U, V> transform)
+    {
+        return configuration.get(configurationKey(key), defaultValue, transform).value();
+    }
+
+    private static String formatKey(final String name, final String key)
+    {
+        return String.format("%s.%s", name, key);
+    }
+
+    /**
+     * Create a configured ElevationUtilities
+     *
+     * @param configuration
+     *            A configuration which should (at a minimum) have a file path for elevation files.
+     */
+    public ElevationUtilities(final Configuration configuration)
```

Comment:
I think this would be more reusable if the 3 srtm values were given as the constructor parameters, instead of needing to be in a configuration.

Reply:
I'll go ahead and make another constructor (I'd like to leave the configuration option in).

Reply:
That sounds good. 
For the configuration constructor, where in the configuration are you expecting the values to be? Are they part of a check object, their own object, or top level entities? It would also be good to have a unit test to make sure custom values are read in properly.

Reply:
I'm expecting them to be part of their own configuration section, e.g.
```json
{
    "elevationutilities": {
        "elevation.srtm_extent": "1" (degree),
        "elevation.srtm_ext": "hgt" (file extension),
        "elevation.path": "elevation"
    }
}
```
(I copied/pasted from `docs/utilities/elevationUtilities`).
I'll see if I can make a specific test for it tomorrow (I thought I had one for that somewhere, but it doesn't look like it).